apps/api/app/utils/notification/orchestrator.py

```
@@ -160,6 +203,21 @@ async def _deliver_notification(self, notification: NotificationRecord) -> None:
             )
 
+    async def _get_channel_prefs(self, user_id: str) -> dict:
+        """Fetch user's notification channel preferences from DB."""
+        try:
+            user = await users_collection.find_one({"_id": ObjectId(user_id)})
+            prefs = (user or {}).get("notification_channel_prefs", {})
+            return {
+                "telegram": prefs.get("telegram", True),
+                "discord": prefs.get("discord", True),
+            }
+        except Exception as e:
```

Ruff (0.15.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)